src/frontend/src/optimizer/plan_node/stream_watermark_filter.rs

```diff
@@ -126,3 +141,23 @@ impl StreamNode for StreamWatermarkFilter {
 
 // TODO(yuhao): may impl a `ExprRewritable` after store `ExplImpl` in catalog.
 impl ExprRewritable for StreamWatermarkFilter {}
+
+struct DisplayWatermarkDesc<'a> {
+    watermark_idx: u32,
+    expr: ExprImpl,
+    input_schema: &'a Schema,
+}
+
+impl fmt::Debug for DisplayWatermarkDesc<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let expr_display = ExprDisplay {
+            expr: &self.expr,
+            input_schema: self.input_schema,
+        };
+        write!(
+            f,
+            "watermark_idx: {}, expr: {}",
```

Comment:
How about making it more compacted?

```suggestion
            "idx: {}, expr: {}",
```